### PR TITLE
Revise update custom field endpoint to indicate that key does change AM-598

### PR DESCRIPTION
**Pull request by JennyAllar**
This PR updates the documentation in the update custom field section to accurately state that the key _will_ change but the name will remain the same.

[Revise API documentation for update-custom-field endpoint to indicate that key does change](https://convertkit.atlassian.net/browse/AM-598)

[T3 that brought this to our attention](https://convertkit.atlassian.net/browse/T3-161)

### CODE REVIEW

**Review by awj — APPROVED** [review]
🙌

File: source/includes/_custom_fields.md [hunk 1]
```diff
@@ -129,7 +129,9 @@ curl -X PUT https://api.convertkit.com/v3/custom_fields/1\
 No content will be returned.
 ```
 
-Updates a custom field label (see Create field above for more information on labels). Note that the key and the name do not change even when the label is updated.
+Updates a custom field label (see Create field above for more information on labels). Note that the key will change but the name remains the same when the label is updated.
+
+_**Warning**: An update to a custom field may break personalization tags in emails._
```
Comment (christinalcole) [inline source/includes/_custom_fields.md:134]: do we know this to be true? if we don't, does it make sense to add it to the docs--versus creating a follow-up issue (e.g.) to determine what actually happens

**Review by christinalcole — APPROVED** [review]
nonblocking comment (b/c the important piece is the changed-key)

File: source/includes/_custom_fields.md (hunk 1 above)
Comment (JennyAllar) [reply source/includes/_custom_fields.md:134]: Great points here with the neutral wording and confustion.  I wonder if we should include it at all. I reached out to @scottjmitchell to get his take as this was added based on a comment he left on the Jira ticket.

**Comment by scottjmitchell** [comment]
Screenshot 2023-04-14 at 18 27 03

It does indeed break liquid relying on custom fields, as updating via the API changes the key and label - so we should make the copy clear to something along the lines of:

_**Warning**: An update to a custom field will break all  of the liquid personalization tags in emails that reference it - e.g. if you updated a `Zip_Code` custom field to become named `Post_Code`, all liquid tags referencing `{{ subscriber.Zip_Code }}` would no longer work and need to be replaced with `{{ subscriber.Post_Code }}` _

This may be a little verbose, but it at least gets the point across!

**Review by christinalcole — APPROVED** [review]
🎉

**Pull request merged**